add draft to guidelines from slack discussion (#2780)
nf-core/website: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md

2. [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the pipeline repository to your GitHub account
3. Make the necessary changes / additions on `dev` branch by using [git checkout](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork) within your forked repository following [pipeline conventions](#pipeline-contribution-conventions)
4. Use `nf-core pipelines schema build` and add any new parameters to the pipeline JSON schema (requires [nf-core tools](https://github.com/nf-core/tools) >= 1.10).

> 5. Submit a Pull Request against the `dev` branch and wait for the code to be reviewed and merged. See below guidelines for this.

Reviewer: I guess this should be changed to point to theother file?

- Fix the bug, and bump version (X.Y.Z+1).
- A PR should be made on `master` from patch to directly resolve this particular bug.

## PR review guidelines

- Authors/Reviewers can request help for review at nf-core Slack `#request-review` channel.
- Anyone can merge after one positive review and with no obvious open questions and, merge ASAP so as not to leave approved PRs hanging.

> - Exception in merges to master for releases (two reviews).

Reviewer 2: Indent these as they're sub-items of the above

Reviewer: ```suggestion
- Exception: merges to master for releases requires **two** reviews.
```

I think the grammar might confuse some people

> - Exception in pseudo PR review for first release (needs review from core or maintainer team).

Reviewer 2: Link to docs for first release

Author: sorry, I cannot find exactly this? I found pipeline release but not first release

Reviewer 3: https://nf-co.re/docs/tutorials/adding_a_pipeline/first_release

> - Exception in pseudo PR review for first release (needs review from core or maintainer team).

Reviewer: ```suggestion
- Exception: pseudo PR reviews for a first release needs **review from core or maintainer team**.
```

> - If it's a major change then it's ok to request a second review - either the author or reviewer can do this.

Reviewer: ```suggestion
- If it's there is a major change in any type of PR (from modules to pipelines) then it's ok to request a second review - either the author or reviewer can do this.
```

> - PRs that have approving reviews but open questions can be merged by the author after addressing the questions to a common-sense level of satisfaction.

Reviewer: ```suggestion
- PRs that have approving reviews but remaining open questions can be merged by the author after addressing the questions to a common-sense level of satisfaction.
```

- If the reviewer feels uneasy about this then they shouldn't leave an approval, but rather a request-changes.
- We prefer to use merge commits to merge. This is to avoid merge conflicts when multiple people are pulling with overlapping feature branches.

> - We prefer verbose commit histories but easy merges.

Reviewer 2: Indent as it's an explanation

- Feature branches should be immediately deleted after merge. Ideally by selecting the new feature on GitHub repo settings

> - Note: squash merging commits in a PR and merging after is fine, that is up to the individuals.

Reviewer 2: Squash merging -> squashing 

Maybe needs a bit more explanation and is a sub point of the merge commit one above

- Fix the bug, and bump version (X.Y.Z+1).
- A PR should be made on `master` from patch to directly resolve this particular bug.

## PR review guidelines

- Authors/Reviewers can request help for review at nf-core Slack `#request-review` channel.
- We encourage collaboration in PR, other contributors are welcome to fix linting errors (you can mention relevant bot commands) or make improvements to the code if necessary.
- Authors/Reviewers should merge after one positive review and with no obvious open questions.
  - Exception: merges to master for releases requires **two** reviews.

>   - Exception: pseudo PR reviews for a first release needs **review from core or maintainer team**.

Reviewer: ```suggestion
  - Exception: pseudo PR reviews for a first release needs **at least one review from core or maintainer team**.
```

>   - If it's there is a major change in any type of PR (from modules to pipelines) then it's ok to request a second review - either the author or reviewer can do this.

Reviewer: ```suggestion
  - If there is a major change in any type of PR (from modules to pipelines) then it's ok to request a second review - either the author or reviewer can do this.
```

  - PRs that have approving reviews but remaining open questions can be merged by the author after addressing the questions to a common-sense level of satisfaction.

>   - If the reviewer feels uneasy about this then they shouldn't leave an approval, but rather a request-changes.

Reviewer: ```suggestion
  - If the reviewer feels uneasy about a remaining open question then they shouldn't leave a only a comment or approval, but rather a request-changes.
```

>   - **request changes** reviews can be [dismissed](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review) by the PR author if considered out of date and resolved.

Reviewer: ```suggestion
  - **request changes** reviews can be [dismissed](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review) by the PR author if considered out of date and resolved.
  - If a PR has comments from one reviewer with open questions, and requests to re-review but does not perform the re-review, the PR may be merged if the first reviewer does not perform the review after 3 months if the PR gets an independent approval from someone else.
```

Something that was discussed in the last maintainers meeting
